addons/crm_livechat/models/crm_lead.py

```diff
@@ -0,0 +1,15 @@
+import json
+from ast import literal_eval
+
+from odoo import fields, models
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    channel_id = fields.Many2one("discuss.channel", "Live chat from which the lead was created", index="btree_not_null")
```

Comment:
This field must be restricted to admin only to avoid giving access to arbitrary channels.

Reply:
Or at least enforce it to be read-only (write override) and only set it during the creation flow?

Reply:
But that's the serpent qui se mord la queue: a high level user can link a channel, which then gives access to the channel to people who can see the lead ?

Reply:
Well admins can do anything so it's not really a problem. If they want to grant lead access to channel its their problem.
But with constraints at least we're limiting admins too.

We're currently considering adding lead_id on channel instead, the code would be cleaner and more robust, but it was functionally possible to be a "many" so we'll have to see how we then have to enforce it to be unique.